python/setup.py

```diff
@@ -256,8 +262,13 @@ def run(self):
             'mllib': ['numpy>=1.7'],
             'sql': [
                 'pandas>=%s' % _minimum_pandas_version,
-                'pyarrow>=%s' % _minimum_pyarrow_version,
-            ]
+                'pyarrow>=1.0.0',
+            ],
+            'pandas_on_spark': [
+                'pandas>=%s' % _minimum_pandas_version,
+                'pyarrow>=0.10',
```

Comment:
The pandas on Spark is based on PySpark SQL with pandas UDF which won't work with `pyarrow < 1.0.0`. We should use the same lower bound.

Reply:
Makes sense. The same lower bound is used.